switch/ffmpeg/tls.patch

```
@@ -0,0 +1,276 @@
++ */
++
++#include <switch.h>
++
++#include "avformat.h"
++#include "internal.h"
++#include "url.h"
++#include "tls.h"
++#include "libavutil/parseutils.h"
++
++typedef struct TLSContext {
++    const AVClass *class;
++    TLSShared tls_shared;
++    SslContext context;
++    SslConnection conn;
++} TLSContext;
++
++static int tls_close(URLContext *h)
++{
++    TLSContext *tls_ctx = h->priv_data;
++
++    sslConnectionClose(&tls_ctx->conn);
++    sslContextClose(&tls_ctx->context);
++
++    ffurl_closep(&tls_ctx->tls_shared.tcp);
++    return 0;
++}
++
++static int tls_open(URLContext *h, const char *uri, int flags, AVDictionary **options)
++{
++    TLSContext *tls_ctx = h->priv_data;
++    TLSShared *shr = &tls_ctx->tls_shared;
++    int ret;
++    Result rc;
++
++    if ((ret = ff_tls_open_underlying(shr, h, uri, options)) < 0)
++        goto fail;
++
++    if (shr->listen) {
++        av_log(h, AV_LOG_ERROR, "TLS Listen Sockets with libnx is not implemented.\n");
++        ret = AVERROR(EINVAL);
++        goto fail;
++    }
++
++    rc = sslCreateContext(&tls_ctx->context, SslVersion_Auto);
++    if (R_FAILED(rc)) {
++        ret = AVERROR(EINVAL);
++        goto fail;
++    }
++
++    rc = sslContextCreateConnection(&tls_ctx->context, &tls_ctx->conn);
++
++    if (R_SUCCEEDED(rc)) {
++        rc = sslConnectionSetOption(&tls_ctx->conn, SslOptionType_DoNotCloseSocket, true);
++    }
++
++    if (R_SUCCEEDED(rc)) {
++        int *sockfd = shr->tcp->priv_data + sizeof(const AVClass *);
++        ret = socketSslConnectionSetSocketDescriptor(&tls_ctx->conn, *sockfd);
++        if (ret == -1 && errno != ENOENT) {
++            av_log(h, AV_LOG_ERROR, "socketSslConnectionSetSocketDescriptor returned: %u\n", rc);
++            goto fail;
++        }
++    }
++
++    if (R_SUCCEEDED(rc)) {
++        rc = sslConnectionSetHostName(&tls_ctx->conn, shr->host, strlen(shr->host));
++    }
++
++    /* This will fail on system-versions where this option isn't available,
```

Is this an outdated comment? You don't appear to be ignoring errors here.

---

fixed error handling